src/anemoi/training/config/dataloader/native_grid.yaml

```diff
@@ -8,9 +8,9 @@ pin_memory: True
 #   which is then all-gathered between the group.
 #   This can reduce CPU memory usage as well as increase dataloader throughput.
 #   The number of GPUs per model must be divisible by read_group_size.
-#   Good values are num_gpus_per_model or num_gpus_per_node.
+#   To disable, set to 1.
 # ============
-read_group_size: 1
+read_group_size: ${hardware.num_gpus_per_model}
```

Comment:
This design choice is good for global, but would break distributed LAMs?

Reply:
I think it should be fine. In our sharding, all model instances expect a full batch of data. Here the batch is gathered before it goes to the model. Do we have a simple test setup for lam?

Reply:
If I am looking at the correct branch, 

https://github.com/ecmwf/anemoi-training/blob/4b564e94a35bd028f9d4f54dab25fbb38bf2c8a5/src/anemoi/training/data/dataset.py#L241C1-L242C47

you working with an index. So instead of self.grid_start / self.grid_end, one could compute indices for each reader, either from the full grid or from the subset.

Reply:
I've added support for LAM in [f08123a](https://github.com/ecmwf/anemoi-training/pull/76/commits/f08123a4f430afd753b2fa08c058a9cb53d5c618). It's a bit weird: anemoi-datasets doesn't support slicing the grid dimension for LAM datasets so I had to load the full grid into memory and then slice the numpy array. This isn't optimal in terms of CPU memory usage but still an improvement. 

We could use this as a preliminary solution and consider updating anemoi-datasets to support slicing LAM data directly in the future.

Reply:
I've opened a small PR in anemoi-datasets (https://github.com/ecmwf/anemoi-datasets/pull/145) to support the slicing datasets with cutout. When that gets merged we can drop the case distinction from [f08123a](https://github.com/ecmwf/anemoi-training/pull/76/commits/f08123a4f430afd753b2fa08c058a9cb53d5c618).